specs/governance/alligator.md

```diff
@@ -0,0 +1,389 @@
+by the [`GovernanceToken`](gov-token.md) contract.
+
+```solidity
+function afterTokenTransfer(address _from, address _to, uint256 _amount) external
+```
+
+The `Alligator` MUST check if the `_from` or `_to` addresses have been migrated by checking the `migrated` mapping
+from its [storage](#storage). If either address has not been migrated, the `Alligator` MUST copy the delegation
+and checkpoint data from the token contract to its own state. After copying the data, the `Alligator` MUST update
+the `migrated` mapping to reflect that the address has been migrated.
+
+The `afterTokenTransfer` function MUST update the voting power of the `_from` and `_to` addresses with the same logic
+of the `GovernanceToken`.
+
+#### `migrateAccount`
+
+Migrates the delegation state of a given account from the `GovernanceToken` to the `Alligator` contract. This function
+MUST only migrate the `_account` if it has not been migrated yet.
+
+```solidity
+function migrateAccount(address _account) external
+```
+
+### Getters
+
+For backwards compatibility, the `Alligator` MUST implement all public getter functions of the
+[`GovernanceToken`](gov-token.md) related to delegation and voting power. These functions MUST be used by the
+[`GovernanceToken`](gov-token.md) when an account has been been migrated to the `Alligator` contract. Otherwise,
+the [`GovernanceToken`](gov-token.md) MUST use its own state.
+
+#### `checkpoints`
+
+Retrieves the checkpoints for a given user address.
+
+```solidity
+function checkpoints(address _account) external view returns (ERC20Votes.Checkpoint[] memory)
+```
+
+#### `numCheckpoints`
+
+Retrieves the number checkpoints for a given user address.
+
+```solidity
+function numCheckpoints(address _account) external view returns (uint32)
+```
+
+#### `getVotes`
+
+Retrieves the voting power of a given user address.
+
+```solidity
+function getVotes(address _account) external view returns (uint256)
+```
+
+#### `getPastVotes`
+
+Retrieves the voting power of a given user address at a block.
+
+```solidity
+function getPastVotes(address _account, uint256 _blockNumber) external view returns (uint256)
+```
+
+#### `getPastTotalSupply`
+
+Retrieves the total supply of the `GovernanceToken` at a given block.
+
+```solidity
+function getPastTotalSupply(uint256 _blockNumber) external view returns (uint256)
+```
+
+#### `subdelegations`
+
+Retrieves the subdelegations for a given user and delegatee.
+
+```solidity
+function subdelegations(address _account, address _delegatee) external view returns (SubdelegationRules memory)
+```
+
+### Events
+
+#### `Subdelegation`
+
+MUST trigger when an account subdelegates voting power to another address (delegatee).
+
+```solidity
+event Subdelegation(address indexed account, address indexed delegatee, SubdelegationRules rule)
+```
+
+#### `Subdelegations`
+
+MUST trigger when an account subdelegates voting power to multiple addresses (delegatees).
```

Comment:
would this be emitting the requested `subdelegations`  or the `subdelegations` state after the changes have been applied?

Reply:
https://github.com/ethereum-optimism/specs/pull/247/commits/88c104f1a1ce5af225d224cbe324c0f63bcd8a5e